### PR TITLE
LibWeb: Shadow Realm integration - part 2

### PR DESCRIPTION
**Pull request by shannonbooth**
Second stage of shadow realm integration into LibWeb sitting ontop of https://github.com/LadybirdBrowser/ladybird/pull/1927

The step after this is to change HTML::Script to operate on realm's instead of environment setting objects. After that, we should then be able to hook up the VM hook to actually initialize the shadow realm global object - which is enough for the test case of:

```html
<script>
async function test() {
    const realm = new ShadowRealm()
    const aSync = realm.evaluate(`async () => 1 + 1`)
    aSync()
}

test();
</script>
```

To not crash :^)

Edit: For a mostly-working implementation, see: https://github.com/LadybirdBrowser/ladybird/pull/1993

### CODE REVIEW

**Comment by ADKaster** [comment]
Pre-review comment: I remember Luke Warlow mentioning in discord that the example there is supposed to throw, do we handle that yet? or is that later :)

**Comment by shannonbooth** [comment]
Pushed rebase to fix de conflict

File: Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp
Comment (ADKaster) [inline Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp:151]: Is there a tracking issue in whatwg/html or the shadow realm repo for updating the other specs to this new 'principal' verbiage? If not, this commit would be a good place to start.

File: Userland/Libraries/LibWeb/Bindings/AudioConstructor.cpp
Comment (ADKaster) [inline Userland/Libraries/LibWeb/Bindings/AudioConstructor.cpp:44]: In the first commit you updated the copied spec text language to add "principal" for the ESO. Seems a miss here, and in Animations/Animation.cpp.

... and more. Perhaps best to leave that to a tracking issue in html/shadow realms

**Review by ADKaster — APPROVED** [review]
This part seems pretty reasonable to me. Two comments about how we're going to follow-up or track changes to other specs, but they don't need to be addressed before getting this set in main, I don't think.

I think after this shadow realm work is in, we can revisit the libgc changes. We'll want that for wasm gc at the very least. And would be best to not be juggling too many sweeping changes at once.

File: Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp:151]: This update in verbiage is covered for the HTML spec in the merge request of https://github.com/whatwg/html/pull/9893

Some of the other specs also have updated verbiage, such as https://github.com/whatwg/webidl/pull/1437

Not all specs have an open merge request with an update (such as the XHR one https://github.com/whatwg/xhr/pulls) but I think that will come once the HTML spec change is merged.

File: Userland/Libraries/LibWeb/Bindings/AudioConstructor.cpp
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/Bindings/AudioConstructor.cpp:44]: Whoops, that is a miss on my end for this one. It is covered in the open merge request, see: https://whatpr.org/html/9893/b8ea975...32ad7a4/media.html#dom-audio

I did a scan over the spec folders for specs which have an open MR, but missed this one since it's in the Bindings folder. Will do a second scan over and adjust

**Pull request merged**
